🐞 - Error after migration to 4.0 : Property 'items' in type 'TuiDataListGroupWrapperComponent<T>' is not assignable to the same property in base type

### Playground Link

_No response_

### Description

I am trying to migrate an application that uses Taiga UI 3.89 to version 4.0.

After running the migrations and resolving the //TODO cases that were added to the code, there are still some errors that seem to come internally from the framework.

When building a production build I get this error:


```
X [ERROR] TS2416: Property 'items' in type "TuiDataListGroupWrapperComponent<T>" is not assignable to the same property in base type "TuiDataListWrapperComponent<T>".
  Type "readonly T[][]" is not assignable to type "readonly T[]".
    Type "T[]" is not assignable to type "T".
      "T" could be instantiated with an arbitrary type which could be unrelated to "T[]". [plugin angular-compiler]

node_modules/.pnpm/@taiga-ui+kit@4.0.0_guvll333atvit5skvvaejqvrwe/node_modules/@taiga-ui/kit/components/data-list-wrapper/data-list-group-wrapper.component.d.ts:4:4:
      4 │     items: readonly T[][] | null;
```


![image](https://github.com/user-attachments/assets/4361bf55-f08d-4b0f-b8b3-d1da231c709a)


### Angular version

18.1.4

### Taiga UI version

4.0

### Which browsers have you used?

- [X] Chrome
- [X] Firefox
- [X] Safari
- [X] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

--- a/projects/kit/components/data-list-wrapper/data-list-group-wrapper.component.ts
+++ b/projects/kit/components/data-list-wrapper/data-list-group-wrapper.component.ts
@@ -16,11 +16,10 @@ import {TuiDataListWrapperComponent} from './data-list-wrapper.component';
     changeDetection: ChangeDetectionStrategy.OnPush,
     providers: [tuiAsDataListAccessor(TuiDataListGroupWrapperComponent)],
 })
-export class TuiDataListGroupWrapperComponent<T> extends TuiDataListWrapperComponent<T> {
-    @Input()
-    // @ts-ignore
-    public override items: readonly T[][] | null = [];
-
+export class TuiDataListGroupWrapperComponent<T> extends TuiDataListWrapperComponent<
+    T,
+    T[]
+> {
     @Input()
     public labels: readonly string[] = [];
 }

--- a/projects/kit/components/data-list-wrapper/data-list-wrapper.component.ts
+++ b/projects/kit/components/data-list-wrapper/data-list-wrapper.component.ts
@@ -36,14 +36,14 @@ import {PolymorpheusOutlet} from '@taiga-ui/polymorpheus';
     changeDetection: ChangeDetectionStrategy.OnPush,
     providers: [tuiAsDataListAccessor(TuiDataListWrapperComponent)],
 })
-export class TuiDataListWrapperComponent<T> {
+export class TuiDataListWrapperComponent<T, K = T> {
     private readonly itemsHandlers: TuiItemsHandlers<T> = inject(TUI_ITEMS_HANDLERS);
 
     @ViewChildren(forwardRef(() => TuiOption))
     protected readonly optionsQuery: QueryList<TuiOption<T>> = EMPTY_QUERY;
 
     @Input()
-    public items: readonly T[] | null = [];
+    public items: readonly K[] | null = [];
 
     @Input()
     public disabledItemHandler: TuiItemsHandlers<T>['disabledItemHandler'] =
@@ -74,5 +74,9 @@ export class TuiDataListWrapperComponent<T> {
             .filter(({disabled}) => includeDisabled || !disabled)
             .map(({value}) => value)
             .filter(tuiIsPresent);
+    }
+
+    protected $cast(items: readonly K[]): readonly T[] {
+        return items as unknown as readonly T[];
     }
 }

--- a/projects/kit/components/data-list-wrapper/data-list-wrapper.template.html
+++ b/projects/kit/components/data-list-wrapper/data-list-wrapper.template.html
@@ -4,7 +4,7 @@
     [size]="size"
 >
     <button
-        *ngFor="let item of items"
+        *ngFor="let item of $cast(items)"
         #elementRef="elementRef"
         automation-id="tui-data-list-wrapper__option"
         tuiElement

